fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/PrettyExplainFormatter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * ExplainFormatter.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades;
+
+import com.apple.foundationdb.record.query.plan.cascades.ExplainTokens.ToStringToken;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+/**
+ * A formatter for tokens.
+ */
+public class PrettyExplainFormatter extends DefaultExplainFormatter {
+    private static final int TAB_SIZE = 4;
+
+    private final boolean useOptionalLineBreaks;
+    private int indentationLevel;
+
+    public PrettyExplainFormatter(@Nonnull final Supplier<ExplainSymbolMap> symbolMapSupplier,
+                                  final boolean useOptionalLineBreaks) {
+        super(symbolMapSupplier);
+        this.useOptionalLineBreaks = useOptionalLineBreaks;
+        this.indentationLevel = 0;
+    }
+
+    @Nonnull
+    @Override
+    public CharSequence visitLineBreakOrSpace(@Nonnull final ExplainTokens.LineBreakOrSpaceToken lineBreakOrSpaceToken,
+                                              @Nonnull final CharSequence stringedToken) {
+        if (useOptionalLineBreaks) {
+            return "\n" + " ".repeat(indentationLevel * TAB_SIZE);
+        }
+        return " ";
+    }
+
+    @Nonnull
+    @Override
+    public CharSequence visitIdentifier(@Nonnull final ExplainTokens.IdentifierToken identifierToken,
+                                  @Nonnull final CharSequence stringedToken) {
+        return new StringBuilder().append(Color.BRIGHT_YELLOW).append(stringedToken).append(Color.RESET);
```

Comment:
I'm not sure what the right way to do this is, but we may want to make these colors configurable, somehow. In particular, I think the choices here are optimized for display against a dark background, but it may be nigh unreadable on a white background (particularly the white text). If we want to defer that to the future, and take the rest of this now, that makes sense to me, as it's pretty easily separable work.

Reply:
Yes, let's please defer that.

Reply:
https://github.com/FoundationDB/fdb-record-layer/issues/3018